torch_xla/csrc/translator.cpp

```diff
@@ -511,10 +537,21 @@ XlaComputationInOut XlaTranslator::BuildComputationProgram(
     AT_ERROR("Unexpected end of graph");
   }
   std::vector<xla::XlaOp> returned_tuple;
-  for (const auto return_input : node_inputs) {
+  XlaComputationInOut::SizeOpValues ret_size_op_values;
+  for (size_t return_input_idx = 0; return_input_idx < node_inputs.size();
+       ++return_input_idx) {
+    const auto return_input = node_inputs[return_input_idx];
+    const auto it = size_op_values_tracking.find(return_input->unique());
+    // Add evaluated aten::size values to the return tuple.
+    if (it != size_op_values_tracking.end()) {
+      const auto it_ok = ret_size_op_values.emplace(
+          std::make_pair(return_input_idx, it->second));
+      XLA_CHECK(it_ok.second);
+    }
     returned_tuple.push_back(cctx.GetOpForValue(return_input));
   }
-  return XlaComputationInOut{cctx.ReleaseInputs(), returned_tuple};
+  return XlaComputationInOut{cctx.ReleaseInputs(), returned_tuple,
```

Comment:
I'd add an explicit std::move(returned_tuple) and have a cctx.ReleaseSizeOpValues().